app/controllers/locations/map_controller.rb

```diff
@@ -0,0 +1,23 @@
+require_dependency "locations/application_controller"
+
+module Locations
+  class MapController < ApplicationController
+
+    def index
+        @map = Map.new(map_params)
+
+        if !@map.url.nil?
```

Comment:
The logic here feels a little confusing, and violates law of demeter. What's really being checked here? If the map is valid? Can that be extracted to `unless @map.valid?`